### PR TITLE
Retrieve the matrix base config using repos

### PR DESCRIPTION
**Pull request by ekohl**
The actions/checkout action is more flexible than recreating this via URLs. It can retrieve a single file from a git clone. This is especially useful where the repository stores the same file and you want to test changes to a pipeline as well as the matrix configuration.

### CODE REVIEW

File: build_matrix [hunk 1]
```diff
@@ -3,13 +3,11 @@
...
 import os
-from urllib.request import urlopen
 from pprint import pprint
 
 
-def main(matrix_url, include, exclude):
-    with urlopen(matrix_url) as response:
-        matrix = json.load(response)
+def main(matrix, include, exclude):
+    matrix = json.load(matrix)
```
Comment (evgeni) [inline build_matrix]: the one matrix is a string, the other a dict, someone gonna yell at you (→ mypy).

**Review by evgeni — APPROVED** [review]
I kinda liked the URL as a fallback, but I guess with the sparse checkout it's as quick as it can get

File: build_matrix (hunk 1 above)
Comment (ekohl) [reply build_matrix]: That's what you get for renaming variables.

File: action.yml [hunk 2]
```diff
@@ -2,9 +2,16 @@
 name: 'Compose a GitHub Actions matrix'
 description: "Combine a base matrix with additional parameters"
 inputs:
-  base_matrix_url:
-    description: 'URL to the file containing the base matrix, encoded as JSON'
-    required: true
+  repository:
+    description: 'Repository to clone'
+    required: false
```
Comment (ofedoren) [inline action.yml:7]: Optional without default value? Wouldn't it crash later?

Comment (ekohl) [reply action.yml:7]: In testing it appears to work as I expected. I think it has a value of `null` and then `actions/checkout` does the right thing.

**Pull request merged**